src/runtime/stream.rs

```diff
@@ -165,12 +161,14 @@ impl AsyncRead for AsyncStream {
     fn poll_read(
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-        buf: &mut [u8],
+        mut buf: &mut [u8],
     ) -> Poll<std::io::Result<usize>> {
         match self.deref_mut() {
             Self::Null => Poll::Ready(Ok(0)),
             Self::Tcp(ref mut inner) => AsyncRead::poll_read(Pin::new(inner), cx, buf),
-            Self::Tls(ref mut inner) => Pin::new(inner).poll_read(cx, buf),
+            Self::Tls(ref mut inner) => {
+                tokio_util::io::poll_read_buf(Pin::new(inner), cx, &mut buf)
```

Comment:
Because tokio's AsyncRead trait was changed to use the new `ReadBuf` type instead of `[u8]`, we need to use an adapter to implement this trait.